src/test/java/com/cloudbees/jenkins/plugins/bitbucket/BranchScanningTest.java

```diff
@@ -133,6 +133,25 @@ public void mercurialSCMTest() throws Exception {
         assertTrue("SCM must be an instance of MercurialSCM", scm instanceof MercurialSCM);
     }
 
+    /**
+     * Tests scanning of Mercurial repos that have a branch name with characters that need to be uri encoded. Testing
+     * with no cred access
+     */
+    @Test
+    public void mercurialBranchNameUriEncodingTest () throws Exception {
+        BitbucketCloudApiClient api = new BitbucketCloudApiClient("alexjo", "bugrep-forest", null);
```

Comment:
This test is kind of breaking the mock-way this plugin is tested, and having a test depending on an external arbitrary repository is not what I would do here.